### DISCUSSION TITLE
Envoy doing duplicate DNS resolution for STRICT_DNS service entries

### DISCUSSION THREAD

**Discussion by nirvanagit** [istio/istio#49296] [Q&A]
We have two SE with unique endpoint addresses.
We observed that Envoy is doing duplicate DNS resolution for one of them.

Is this expected, or is this a bug?

FYI: We have 5k STRICT_DNS SE in our clusters, and due to duplicate DNS resolutions we are facing some DNS related issues.

**Comment by howardjohn** [comment]
What does "duplicate DNS resolution" mean

**Reply by nirvanagit** [reply]
There are 2 service entries, and it is observed the Envoy is intermittently making two DNS calls for one of the entries.

It could have been a retry from the c-ares library, but looking at tcpdump did not find any traces of a request which did not have a corresponding response, or if the response was invalid.

**Reply by nirvanagit** [reply]
c-ares library does have retries:
```ARES_OPT_TRIES int tries;
The number of tries the resolver will try contacting each name server before giving up. The default is three tries.
```

But it is not clear if these are retries, and if they are, why so. As didn't find any traces in tcpdump as mentioned earlier.

**Comment by nirvanagit** [comment]
c-ares has a default of 3 tries, (excluding the first attempt)
